File: Data Structures & Algorithms/palindromic-substrings/submission-2.java
class Solution {
    public int countSubstrings(String S) {
        int count = 0;
        char[] s = S.toCharArray();
        int n = s.length;
        
        // 奇數 
        for (int i = 0 ; i < n ; i++) {
            int l = i;
            int r = i;
            while (l >= 0 && r < n) {
                if (s[l] == s[r]) {
                    count++;
                } else {
                    break;
                }
                l--;
                r++;
            }
        }

        // 偶數 
        for (int i = 0 ; i < n ; i++) {
            int l = i;
            int r = i + 1;
            while (l >= 0 && r < n) {
               if (s[l] == s[r]) {
                    count++;
                } else {
                    break;
                }
                l--;
                r++;
            }
        }
        return count;
    }
}

File: Data Structures & Algorithms/hand-of-straights/submission-2.py
class Solution:
    def isNStraightHand(self, hand: List[int], groupSize: int) -> bool:
        if len(hand) % groupSize != 0:
            return False
        count = Counter()
        for num in hand:
            count[num] += 1
        
        while count:
            minKey = min(count.keys())
            for i in range(minKey, minKey + groupSize):
                count[i] -= 1
                if count[i] < 0:
                    return False
                if count[i] == 0:
                    del count[i]    
        return True            
                
